sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,354 @@
+                buildConfiguration));
+            policies.add(new RequestIdPolicy());
+            policies.add(new AddHeadersPolicy(headers));
+            policies.add(new AddDatePolicy());
+
+            HttpPolicyProviders.addBeforeRetryPolicies(policies);
+            policies.add(retryPolicy == null ? DEFAULT_RETRY_POLICY : retryPolicy);
+            policies.addAll(this.policies);
+            HttpPolicyProviders.addAfterRetryPolicies(policies);
+
+            policies.add(new HttpLoggingPolicy(httpLogOptions));
+
+            httpPipeline = new HttpPipelineBuilder()
+                .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(httpClient)
+                .build();
+        }
+
+        final TextAnalyticsClientImpl textAnalyticsAPI = new TextAnalyticsClientImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(httpPipeline)
+            .build();
+
+        return new TextAnalyticsAsyncClient(textAnalyticsAPI, serviceVersion);
+    }
+
+    /**
+     * Set the default client option for one client.
+     *
+     * @param clientOptions TextAnalyticsClientOptions model that includes
+     * {@link TextAnalyticsClientOptions#getDefaultLanguage() default language} and
+     * {@link TextAnalyticsClientOptions#getDefaultCountryHint() default country hint}
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder clientOptions(TextAnalyticsClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    /**
+     * Sets the service endpoint for the Azure Text Analytics instance.
+     *
+     * @param endpoint The URL of the Azure Text Analytics instance service requests to and receive responses from.
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws NullPointerException if {@code endpoint} is null
+     * @throws IllegalArgumentException if {@code endpoint} cannot be parsed into a valid URL.
+     */
+    public TextAnalyticsClientBuilder endpoint(String endpoint) {
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL", ex));
+        }
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the credential to use when authenticating HTTP requests for this TextAnalyticsClientBuilder.
+     *
+     * @param subscriptionKey subscription key
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws NullPointerException If {@code subscriptionKey} is {@code null}
+     */
+    public TextAnalyticsClientBuilder subscriptionKey(String subscriptionKey) {
+        Objects.requireNonNull(subscriptionKey, "'subscriptionKey' cannot be null.");
+        this.subscriptionKey = subscriptionKey;
```

Comment:
Should this set `tokenCredential` to null? Right now if I set a `TokenCredential` for the first client I construct, then want to use a `subscriptionKey` all subsequent clients will use the `TokenCredential`.